search/src/main/java/com/yas/search/kafka/consumer/ProductSyncDataConsumer.java

```diff
@@ -0,0 +1,59 @@
+package com.yas.search.kafka.consumer;
+
+import com.yas.commonlibrary.kafka.cdc.BaseCdcConsumer;
+import com.yas.commonlibrary.kafka.cdc.RetrySupportDql;
+import com.yas.commonlibrary.kafka.cdc.message.ProductCdcMessage;
+import com.yas.search.service.ProductSyncDataService;
+import jakarta.validation.Valid;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.handler.annotation.Headers;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Service;
+
+/**
+ * Product synchronize data consumer for elasticsearch.
+ */
+@Service
+public class ProductSyncDataConsumer extends BaseCdcConsumer<ProductCdcMessage> {
+
+    private final ProductSyncDataService productSyncDataService;
+
+    public ProductSyncDataConsumer(ProductSyncDataService productSyncDataService) {
+        this.productSyncDataService = productSyncDataService;
+    }
+
+    @KafkaListener(
+        id = "product-sync-es",
+        groupId = "product-sync-search",
+        topics = "${product.topic.name}",
+        containerFactory = "productCdcListenerContainerFactory"
+    )
+    @RetrySupportDql(listenerContainerFactory = "productCdcListenerContainerFactory")
+    public void processMessage(
+        @Payload(required = false) @Valid ProductCdcMessage productCdcMessage,
+        @Headers MessageHeaders headers
+    ) {
+        processMessage(productCdcMessage, headers, this::sync);
+    }
+
+    public void sync(ProductCdcMessage productCdcMessage) {
+        if (productCdcMessage.getAfter() != null) {
```

Comment:
In which case the `productCdcMessage.getAfter()` will be null? I remember that it happens in case of a DELETE operation, but could you double check?

Reply:
Currently, the product does not support hard delete, so I assume that `productCdcMessage.getAfter()` will never be null. I will add `@NotNull` to filter out messages where 'after' is null in case.

Reply:
Hi @duylv27,

> Currently, the product does not support hard delete, so I assume that productCdcMessage.getAfter() will never be null.

It's inconsistent since we assume `productCdcMessage.getAfter()` will never be null while we are handling DELETE operation. I think we should treat it as an exception instead of calling `productSyncDataService.deleteProduct(productId);` and cover the logical deletion in the UPDATE operation.
```
                case DELETE:
                    productSyncDataService.deleteProduct(productId);
                    break;
```

Reply:
This `case DELETE` seems unnecessary in both search and recommendation consumers, as we are handling soft delete in update method (the recommendation consumer already follows this approach, while the search consumer does not yet). 

Therefore, I think we should remove the `case DELETE` from the `switch` block then messages with `DELETE` operation will be on dead-letter queue (due to validation fail).

Regarding `elasticsearch`, we will create an external ticket to handle soft delete case.

Do you have any further suggestions? If not, I will proceed with these updates.

Reply:
I'm okay with your suggestions. Thanks, @duylv27